bundler/lib/bundler/dep_proxy.rb

```diff
@@ -4,18 +4,27 @@ module Bundler
   class DepProxy
     attr_reader :__platform, :dep
 
+    @proxies = {}
+
+    def self.get_proxy(dep, platform)
+      @proxies[[dep, platform].hash] ||= new(dep, platform)
+    end
+
     def initialize(dep, platform)
       @dep = dep
       @__platform = platform
     end
 
-    def hash
-      @hash ||= [dep, __platform].hash
+    private_class_method :new
+    private :dup
```

Comment:
This is to avoid potential future programming mistakes where bundler code tries to instantiate or dup `DepProxy` objects directly, right? Should we add a couple of unit tests to cover those code paths and illustrate this?

Reply:
Done in e2573800b5357949ff7be1e6c8afd8cd9e05deaf